concordium-node/src/grpc2.rs

```diff
@@ -1086,7 +1086,22 @@ pub mod server {
                     }
                 });
                 let service = service::queries_server::QueriesServer::new(server);
-                let log_layer = tower_http::trace::TraceLayer::new_for_grpc();
+                let log_layer = {
+                    use tower_http::classify::GrpcCode as Code;
+                    // Change the trace level for gRPC status codes
+                    let classifier = tower_http::classify::GrpcErrorsAsFailures::new()
+                        .with_success(Code::Cancelled)
+                        .with_success(Code::InvalidArgument)
+                        .with_success(Code::NotFound)
+                        .with_success(Code::AlreadyExists)
+                        .with_success(Code::PermissionDenied)
+                        .with_success(Code::Unimplemented)
+                        .with_success(Code::Unavailable)
+                        .with_success(Code::Unauthenticated);
+                    tower_http::trace::TraceLayer::new(tower_http::classify::SharedClassifier::new(
+                        classifier,
+                    ))
```

Comment:
Possibly we should set the logging for failures lower than ERROR. I would suggest even DEBUG is high enough. Reasons:

- Generally, failure codes don't indicate serious errors.
- If a serious error in processing a request occurs, usually there's an ERROR log message generated anyway.
- The log messages generated on failures look like `[ERROR tower_http::trace::on_failure] response failed classification=Code: 12 latency=0 ms` - which is not really very useful.

Reply:
I have now changed it to make every error status code be logged at debug level.